File: full-vineflower/com/htmake/reader/utils/SpringContextUtils.java
package com.htmake.reader.utils;

import org.springframework.beans.BeansException;
import org.springframework.context.ApplicationContext;
import org.springframework.context.ApplicationContextAware;
import org.springframework.stereotype.Component;

@Component
public class SpringContextUtils implements ApplicationContextAware {
   private static ApplicationContext applicationContext;

   @Override
   public void setApplicationContext(ApplicationContext context) throws BeansException {
      applicationContext = context;
   }

   public static ApplicationContext getApplicationContext() {
      return applicationContext;
   }

   public static Object getBean(String name) {
      return applicationContext != null ? getApplicationContext().getBean(name) : null;
   }

   public static <T> T getBean(Class<T> clazz) {
      return applicationContext != null ? getApplicationContext().getBean(clazz) : null;
   }

   public static <T> T getBean(String name, Class<T> clazz) {
      return applicationContext != null ? getApplicationContext().getBean(name, clazz) : null;
   }
}

File: best-of-3/business/com/htmake/reader/utils/SpringContextUtils.kt
/** Business rewrite from reader-pro-3.2.14.jar — readability / audit. */

package com.htmake.reader.utils

import org.springframework.context.ApplicationContext
import org.springframework.context.ApplicationContextAware
import org.springframework.stereotype.Component

@Component
class SpringContextUtils : ApplicationContextAware {
    override fun setApplicationContext(applicationContext: ApplicationContext) {
        ctx = applicationContext
    }
    companion object {
        private lateinit var ctx: ApplicationContext
        @JvmStatic fun <T> getBean(clazz: Class<T>): T = ctx.getBean(clazz)
        @JvmStatic fun <T> getBean(name: String, clazz: Class<T>): T = ctx.getBean(name, clazz)
    }
}
